src/library_pthread.js

```diff
@@ -130,7 +130,22 @@ var LibraryPThread = {
 
       var numWorkersLoaded = 0;
       for (var i = 0; i < numWorkers; ++i) {
+#if EMBED_PTHREAD_MAIN_JS
+        var pthreadMainCode = {{{ PTHREAD_MAIN_CODE }}};
```

Comment:
let's do the `emcc.py` changes in here, instead of adding `PTHREAD_MAIN_CODE` to `settings.js` and then passing it around. We can read it directly here inside a `{{{ }}}`, using `read()`.